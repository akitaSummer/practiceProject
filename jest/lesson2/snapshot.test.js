import { generateConfig, generateAnotherConfig } from "./snapshot";

// 按w后按u可以更新所有快照
test("测试generateConfig函数", () => {
  expect(generateConfig()).toMatchSnapshot({
    // 设置time属性为Date
    time: expect.any(Date)
  });
});

// 按w后按i可以逐个更新所有快照
test("测试generateAnotherConfig函数", () => {
  expect(generateAnotherConfig()).toMatchSnapshot({
    time: expect.any(Date)
  });
});

test("测试generateAnotherConfig函数", () => {
  // 安装prettier后可以使用toMatchInlineSnapshot, 使快照在test文件里出现
  expect(generateAnotherConfig()).toMatchInlineSnapshot(
    {
      time: expect.any(Date)
    },
    `
    Object {
      "domain": "localhost",
      "port": 8080,
      "server": "http://localhost",
      "time": Any<Date>,
    }
  `
  );
});
