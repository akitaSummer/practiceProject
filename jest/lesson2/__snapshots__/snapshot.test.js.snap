// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试generateAnotherConfig函数 1`] = `
Object {
  "domain": "localhost",
  "port": 8080,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;

exports[`测试generateConfig函数 1`] = `
Object {
  "domain": "localhost",
  "port": 8080,
  "server": "http://localhost",
  "time": Any<Date>,
}
`;
